### PR TITLE
Added `on_span_finish` callback

### PR DESCRIPTION
**Pull request by alexandrz**
I needed it, as I store the span in a thread local variable and need to clear it after span is finished. Would you merge this?

### CODE REVIEW

File: lib/rack/tracer.rb [hunk 1]
```diff
@@ -11,14 +11,16 @@ class Tracer
     # @param tracer [OpenTracing::Tracer] A tracer to be used when start_span, and extract
     #        is called.
     # @param on_start_span [Proc, nil] A callback evaluated after a new span is created.
+    # @param on_span_finish [Proc, nil] A callback evaluated after a span is finished.
```
Comment (indrekj) [inline lib/rack/tracer.rb]: Lets use `on_finish_span` to be consistent with `on_start_span`

File: README.md [hunk 2]
```diff
@@ -23,9 +23,11 @@ You can access the created span using
 env['rack.span']
 ```
 
-You can also add a start span callback
+You can also add start and finish span callbacks
 ```ruby
-use Rack::Tracer, on_start_span: Proc.new {|span| do_something(span) }
+use Rack::Tracer,
+    on_start_span: ->(span) { Thread.current[:root_span] = span },
+    on_start_finish: ->(_span) { Thread.current[:root_span] = nil }
```
[the remark this answers is not in the document]
Comment (alexandrz) [reply README.md]: fixed

File: lib/rack/tracer.rb [hunk 3]
```diff
@@ -11,14 +11,16 @@ class Tracer
...
     # @param errors [Array<Class>] An array of error classes to be captured by the tracer
     #        as errors. Errors are **not** muted by the middleware, they're re-raised afterwards.
-    def initialize(app, tracer: OpenTracing.global_tracer, on_start_span: nil, trust_incoming_span: true, errors: [StandardError])
+    def initialize(app, tracer: nil, on_start_span: nil, on_span_finish: nil, trust_incoming_span: true, errors: nil)
```
Comment (indrekj) [inline lib/rack/tracer.rb]: why `errors` change?

File: lib/rack/tracer.rb (hunk 1 above)
Comment (alexandrz) [reply lib/rack/tracer.rb]: also fixed

File: lib/rack/tracer.rb (hunk 3 above)
Comment (indrekj) [reply lib/rack/tracer.rb]: and `tracer` as well. Lets please keep them as they were, as they are not related to this PR

Comment (alexandrz) [reply lib/rack/tracer.rb]: Just moved default options into the method body to make the method definition shorter and easier to read

File: lib/rack/tracer.rb [hunk 4]
```diff
@@ -54,7 +54,11 @@ def call(env)
       span.log(event: 'error', :'error.object' => e)
       raise
     ensure
-      span.finish
+      begin
+        span.finish
+      ensure
+        @on_start_finish.call(span) if @on_start_finish
```
Comment (indrekj) [inline lib/rack/tracer.rb]: typo: `on_start_finish`

Comment (indrekj) [reply lib/rack/tracer.rb]: please also add tests

Comment (alexandrz) [reply lib/rack/tracer.rb]: ok

**Comment by alexandrz** [comment]
Fixed typos and added tests

**Comment by indrekj** [comment]
Released it as 0.5.0

**Comment by alexandrz** [comment]
Thank you!

**Pull request merged**